plugin/pkg/scheduler/schedulercache/node_info.go

```diff
@@ -248,11 +248,13 @@ func (n *NodeInfo) NonZeroRequest() Resource {
 }
 
 // AllocatableResource returns allocatable resources on a given node.
-func (n *NodeInfo) AllocatableResource() Resource {
+// returns a reference of item Resource, for potential update for pluginResources
+// TODO: find better way to handle pluginResources update
+func (n *NodeInfo) AllocatableResource() *Resource {
```

Comment:
Instead of changing this function to return Resource pointer, I would suggest we keep it as is but add a SetAllocatableResource() function.